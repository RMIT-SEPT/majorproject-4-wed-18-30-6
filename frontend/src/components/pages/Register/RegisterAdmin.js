import React from "react";

const RegisterAdmin = () => {
  return (
    <div>
      <p>RegisterAdmin</p>
    </div>
  );
};

export default RegisterAdmin;
