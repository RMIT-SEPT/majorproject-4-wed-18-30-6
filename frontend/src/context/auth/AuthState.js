import React, { useReducer } from "react";
import AuthContext from "./authContext";
import AuthReducer from "./authReducer";

import {
  REGISTER_SUCCESS,
  REGISTER_FAIL,
  LOGIN_SUCCESS,
  LOGIN_FAIL,
  LOGOUT,
  USER_LOADED,
  AUTH_ERROR,
  CLEAR_ERRORS,
} from "../types";

const AuthState = (props) => {
  const initialState = {
    isAuthenticated: false,
    user: null,
    loading: true,
    error: null,
  };

  const [state, dispatch] = useReducer(AuthReducer, initialState);

  const loadUser = async () => {
    // localStorage.token

    try {
      dispatch({
        type: USER_LOADED,
        payload: {
          type: "admin",
        },
      });
    } catch (err) {
      dispatch({
        type: AUTH_ERROR,
        payload: {},
      });
    }
  };

  const registerAdmin = async (formData) => {
    try {
      dispatch({
        type: REGISTER_SUCCESS,
        payload: {},
      });
    } catch (err) {
      dispatch({
        type: REGISTER_FAIL,
        payload: {},
      });
    }
  };

  const registerCustomer = async (formData) => {
    try {
      dispatch({
        type: REGISTER_SUCCESS,
        payload: {},
      });
    } catch (err) {
      dispatch({
        type: REGISTER_FAIL,
        payload: {},
      });
    }
  };

  const registerEmployee = async (formData) => {
    try {
      dispatch({
        type: REGISTER_SUCCESS,
        payload: {},
      });
    } catch (err) {
      dispatch({
        type: REGISTER_FAIL,
        payload: {},
      });
    }
  };

  const login = async (formData) => {
    try {
      dispatch({
        type: LOGIN_SUCCESS,
        payload: {},
      });
    } catch (err) {
      dispatch({
        type: LOGIN_FAIL,
        payload: {},
      });
    }
  };

  const logout = async () => {
    dispatch({
      type: LOGOUT,
    });
  };

  const clearErrors = async () => {
    dispatch({
      type: CLEAR_ERRORS,
    });
  };

  return (
    <AuthContext.Provider
      value={{
        token: state.token,
        isAuthenticated: state.isAuthenticated,
        loading: state.loading,
        error: state.error,
        user: state.user,
        loadUser,
        registerAdmin,
        registerCustomer,
        registerEmployee,
        login,
        logout,
        clearErrors,
      }}
    >
      {props.children}
    </AuthContext.Provider>
  );
};

export default AuthState;
