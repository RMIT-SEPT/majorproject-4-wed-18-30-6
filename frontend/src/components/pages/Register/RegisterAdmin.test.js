import React from "react";
import RegisterAdmin from "./RegisterAdmin"
import {shallow, mount} from "enzyme";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import AuthState from "../../../context/auth/AuthState";

Enzyme.configure({ adapter: new Adapter()});

describe("Register Admin tests", () => {
  
  //Page rendering
  it('should render the page correctly', () => {
    const component = shallow(<AuthState><RegisterAdmin /></AuthState>);
    expect(component).toHaveLength(1);
  });

  //Empty input fields
  it('should have empty input fields for name, address, mobile, email on start', () => {
    const component = mount(<AuthState><RegisterAdmin /></AuthState>);
    expect(component.find('input#name').text()).toEqual('');
    expect(component.find('input#address').text()).toEqual('');
    expect(component.find('input#mobile').text()).toEqual('');
    expect(component.find('input#email').text()).toEqual('');
  });
  
  //Button should call function on-click
  it('should call a function when the button is clicked', () => {
    const component = mount(<AuthState><RegisterAdmin /></AuthState>);
    const mockHandler = jest.fn();
    component.find('register-btn');
    component.simulate('click');
    expect(mockHandler.mock.calls.length).toBe(1);
  });
  
  //test input fields
  it('should input correctly', () => {
    const component = mount(<AuthState><RegisterAdmin /></AuthState>);
    const text = component.find('input#name');
    text.instance().name = 'dummy';
    text.simulate('change');
    expect(text.instance().name).toEqual('dummy');
  });
});