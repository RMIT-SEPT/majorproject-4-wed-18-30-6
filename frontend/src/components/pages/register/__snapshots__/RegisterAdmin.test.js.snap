// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Admin tests should render correctly 1`] = `
<div
  className="register"
>
  <h2
    className="transform"
  >
    <span
      className="big"
    >
      A
    </span>
    <span
      className="small"
    >
      dmin 
    </span>
    <span
      className="big"
    >
      S
    </span>
    <span
      className="small"
    >
      ign 
    </span>
    <span
      className="big"
    >
       U
    </span>
    <span
      className="small"
    >
      p
    </span>
  </h2>
  <h3>
    Please fill in the form
  </h3>
  <form
    onSubmit={[Function]}
  >
    <div
      className="input-container"
    >
      <span
        className="header"
        htmlFor="name"
      >
        Name
      </span>
      <input
        className="input"
        id="name"
        name="name"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="input-container"
    >
      <span
        className="header"
        htmlFor="address"
      >
        Address
      </span>
      <input
        className="input"
        id="address"
        name="address"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div>
      <div
        className="input-container"
        style={
          Object {
            "display": "inline-block",
            "width": "50%",
          }
        }
      >
        <span
          className="header"
          htmlFor="mobile"
        >
          Mobile
        </span>
        <input
          className="input"
          id="mobile"
          name="mobile"
          onChange={[Function]}
          type="phone"
        />
      </div>
      <div
        className="input-container"
        style={
          Object {
            "display": "inline-block",
            "width": "50%",
          }
        }
      >
        <span
          className="header"
          htmlFor="email"
        >
          Email
        </span>
        <input
          className="input"
          id="email"
          name="email"
          onChange={[Function]}
          type="email"
        />
      </div>
    </div>
    <div>
      <div
        className="input-container"
        style={
          Object {
            "display": "inline-block",
            "width": "50%",
          }
        }
      >
        <span
          className="header"
          htmlFor="skills"
        >
          Skills
        </span>
        <select
          className="input"
          id="skills"
          name="skills"
          onChange={[Function]}
        >
          <option
            value="cleaning"
          >
            Cleaning
          </option>
          <option
            value="sweeping"
          >
            Sweeping
          </option>
        </select>
      </div>
      <div
        className="input-container"
        style={
          Object {
            "display": "inline-block",
            "width": "50%",
          }
        }
      >
        <span
          className="header"
          htmlFor="type"
        >
          Type
        </span>
        <select
          className="input"
          id="type"
          name="type"
          onChange={[Function]}
        >
          <option
            value="tier1"
          >
            Tier1
          </option>
          <option
            value="tier2"
          >
            Tier2
          </option>
        </select>
      </div>
    </div>
    <div
      className="input-container"
    >
      <span
        className="header"
        htmlFor="availability"
      >
        Availability
      </span>
      <input
        className="input"
        id="availability"
        name="availability"
        onChange={[Function]}
        type="text"
      />
    </div>
    <input
      className="register-btn"
      type="submit"
      value="Sign up"
    />
  </form>
  <h3
    style={
      Object {
        "position": "relative",
        "top": "90px",
      }
    }
  >
    Your request will be sent to the management for approval
  </h3>
</div>
`;
